build.gradle

```diff
@@ -91,7 +91,7 @@ if (!usingRemoteCluster) {
 }
 
 ext {
-    opensearchVersion = '1.0.0'
+    opensearchVersion = '1.1.0-SNAPSHOT'
```

Comment:
This is no longer used.

Reply:
I see reference here: https://github.com/opensearch-project/k-NN/blob/8b02dca183e383563562c5a419bc38e2f69e5e24/build.gradle#L280

Can this be replaced with `opensearch_version`?

Reply:
I believe so yes!

Reply:
Replace this with `opensearch_build`.

Reply:
You could do something like this: https://github.com/opensearch-project/anomaly-detection/pull/176/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R36

Or `opensearch_version - '-SNAPSHOT' + '.0', but this is about knnlib. So what version should knnlib be? I imagine it should also be a `-SNAPSHOT` version? In which case this is just `${project.version}`?

@jmazanec15 can comment maybe?

Reply:
Right, I believe right now the library version [defaults](https://github.com/opensearch-project/k-NN/blob/main/jni/CMakeLists.txt#L44) to "local". The "requires" task ensures that when installing deb/rpm, the library with version "${opensearchVersion}.0" is installed (So I believe RPM/DEB installation may fail unless the cmake env variable is updated @VijayanB can you comment on this?).

Because the library is packaged separately from the plugin, I think the library dependency version could be treated as a separate variable. I do not think we need to have snapshot for the library -- we could just hardcode the library version.

Reply:
Ok, so I think the ask for this PR is 

1. Add support for `-Dopensearch.knnlib.version=`.
2. Default it to 1.1.0.0? Or ... where is that set and what is the version right now? 
3. Adjust the requires to use this value.

Reply:
@jmazanec15 i believe knn lib version should be set to knn plugin version. 
@dblock will project.version == 1.0.0.0-SNAPSHOT ?

Reply:
No, it will be 1.1.0.0-SNAPSHOT for snapshot builds, and 1.1.0.0 for dailies.

Reply:
got it. I propose to use project.version. @jmazanec15 any concerns?

Reply:
Note -- this just impacts RPM/DEB creation I believe. 

So, at the moment, the library packages (RPM/DEB) are built outside of the build.gradle process. The library packages, by default, have "local" as the version, unless otherwise specified. 

I think we should:
(1) Change default JNI version to "1.1.0.0" (In Cmake)
(2) Hardcode "1.1.0.0" as the library version here.

In a future PR, we can figure out a better way to create snapshots of the library.

Reply:
This way, we treat library as a separate dependency, which has its own versioning.

Reply:
made changes as propose. @jmazanec15 can you review again?